com.unity.render-pipelines.universal/Runtime/Passes/RenderObjectsPass.cs

```diff
@@ -120,11 +121,31 @@ public override void Execute(ScriptableRenderContext context, ref RenderingData
                     }
                 }
 
-                context.ExecuteCommandBuffer(cmd);
-                cmd.Clear();
+                if((DebugHandler != null) && DebugHandler.IsActiveForCamera(ref cameraData))
+                {
+                    foreach(DebugRenderSetup debugRenderSetup in DebugHandler.CreateDebugRenderSetupEnumerable(context, cmd))
+                    {
+                        DrawingSettings debugDrawingSettings = debugRenderSetup.CreateDrawingSettings(ref renderingData, drawingSettings);
+
+                        if(debugRenderSetup.GetRenderStateBlock(out RenderStateBlock renderStateBlock))
+                        {
+                            context.DrawRenderers(renderingData.cullResults, ref debugDrawingSettings, ref m_FilteringSettings, ref renderStateBlock);
+                        }
+                        else
+                        {
+                            context.DrawRenderers(renderingData.cullResults, ref debugDrawingSettings, ref m_FilteringSettings, ref m_RenderStateBlock);
+                        }
+                    }
+                }
+                else
+                {
+                    // Ensure we flush our command-buffer before we render...
+                    context.ExecuteCommandBuffer(cmd);
+                    cmd.Clear();
 
-                context.DrawRenderers(renderingData.cullResults, ref drawingSettings, ref m_FilteringSettings,
-                    ref m_RenderStateBlock);
+                    // Render the objects...
+                    context.DrawRenderers(renderingData.cullResults, ref drawingSettings, ref m_FilteringSettings, ref m_RenderStateBlock);
+                }
```

Comment:
Seems this logic is repeated code in several passes and also if users would need to add custom pass they would have to know about this logic as well. I wonder if we could encapsulate this, maybe with a `DrawRenderers` utility?